java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java

```diff
@@ -678,7 +678,16 @@ static ObjectPrx checkedCast(ObjectPrx obj) {
    * @return <code>obj</code>.
    */
   static ObjectPrx checkedCast(ObjectPrx obj, java.util.Map<String, String> context) {
-    return obj;
+    if (obj != null) {
+      try {
+        boolean ok = obj.ice_isA(ice_staticId, context);
+        if (ok) {
+          return new _ObjectPrxI(obj);
```

Comment:
Why do we need to call the constructor here?

What about doing:

```
static ObjectPrx checkedCast(ObjectPrx obj, java.util.Map<String, String> context) {
    if (obj != null) {
      try {
        return obj.ice_isA(ice_staticId, context) ? obj : null;
      } catch (FacetNotExistException ex) {
      }
    }
    return null;
  }
```

Reply:
@pepone suggestion is what we do in C# - minus the FacetNotExist handling:
https://github.com/zeroc-ice/ice/blob/948be92fe711ad03d4043c60d24503433c588226/csharp/src/Ice/Proxy.cs#L1828

Reply:
Yes, you're right, there's no need to create a new proxy here.
Changed as suggested.